stl/inc/xstring

```diff
@@ -3992,7 +3992,7 @@ public:
             return;
         }
 
-        const size_type _Target_capacity = (_STD min)(_My_data._Mysize | _ALLOC_MASK, max_size());
+        size_type _Target_capacity = (_STD min)(_My_data._Mysize | _ALLOC_MASK, max_size());

```

Comment:
Hmm, perhaps irrelevant to the test_failure, but I realize that after the change `_Allocate_for_capacity` should not apply to `shrink_to_fit` here.

Reply:
On second thought, I think this change is not as harmful as it appears, as `shrink_to_fit` `"is a non-binding request to reduce capacity() to size()"`, and in my understanding a replacement of `allocate` to `allocate_at_least` will not result in extra allocation(maybe I'm wrong here) - it just reflect a situation that the string can end up `taking up` more space(whether usable by the string) after "shrinking"

Reply:
I don't think we should mess with `shrink_to_fit`'s behavior, and have pushed a commit to imitate `vector`'s exact policy.